src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/DateTimeFilter.php

```
@@ -0,0 +1,255 @@
+                FieldFilterHelper::checkDateTime($field, $value, static::DATETIME_FORMAT, static::class);
+
+                break;
+            case Operators::BETWEEN:
+            case Operators::NOT_BETWEEN:
+                if (!is_array($value)) {
+                    throw InvalidPropertyTypeException::arrayExpected($field, static::class, $value);
+                }
+
+                if (2 !== count($value)) {
+                    throw InvalidPropertyTypeException::validArrayStructureExpected(
+                        $field,
+                        sprintf('should contain 2 strings with the format "%s"', static::DATETIME_FORMAT),
+                        static::class,
+                        $value
+                    );
+                }
+
+                foreach ($value as $singleValue) {
+                    FieldFilterHelper::checkDateTime($field, $singleValue, static::DATETIME_FORMAT, static::class);
+                }
+
+                break;
+            case Operators::SINCE_LAST_JOB:
+                if (!is_string($value)) {
+                    throw InvalidPropertyTypeException::stringExpected($field, static::class, $value);
+                }
+
+                break;
+            case Operators::SINCE_LAST_N_DAYS:
+                if (!is_numeric($value)) {
+                    throw InvalidPropertyTypeException::numericExpected($field, static::class, $value);
+                }
+
+                break;
+            case Operators::IS_EMPTY:
+            case Operators::IS_NOT_EMPTY:
+                break;
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+    }
+
+    /**
+     * @param string|\DateTime $value
+     *
+     * @return int
+     */
+    protected function getFormattedDate($value) {
+        if (!$value instanceof \DateTime) {
+            $value = \DateTime::createFromFormat(static::DATETIME_FORMAT, $value);
```

it can be false
and in that case we should throw an exception

---

remember to add a spec for each exception that you throw ;)